### PR TITLE
fix: initials account image for dark mode - WPB-11833

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11833" title="WPB-11833" target="_blank">BugWPB-11833</a>  [iOS] Dark mode support for account image generation
  </td></table>
  <br />
 

### Issue

Currently the placeholder account image (the user's initials) doesn't display correctly in dark mode. Also the image doesn't update when switching the user interface style.

This PR moves the use case from WireUI into WireDataModel.
Also the AccountImageView supports a text (initials) and an image value now.

### Testing

- Open the app in light and dark mode.
- While the app is open, switch from light to dark and vice versa.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [x] Make sure you use the API for UI elements that support large fonts.
- [x] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [x] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios-data-model/Tests/UseCases/AccountImageSource/GetTeamAccountImageSourceUseCaseTests.swift [hunk 1]
```diff
@@ -16,32 +16,34 @@
...
 
-import WireFoundation
+import WireDataModelSupport
 import XCTest
 
-@testable import WireAccountImageUI
+@testable import WireDataModel
 
-final class GetTeamAccountImageUseCaseTests: XCTestCase {
+// TODO: finish implementation
```
Comment (caldrian) [inline wire-ios-data-model/Tests/UseCases/AccountImageSource/GetTeamAccountImageSourceUseCaseTests.swift:24]: I will finish it in a moment.

File: WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/textMode_available.dark.png
Comment (samwyndham) [inline WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/textMode_available.dark.png:1]: The text in these files is pixelated? Is this a bug?

File: wire-ios-data-model/Source/UseCases/AccountImageSource/AccountImageSource.swift [hunk 2]
```diff
@@ -16,13 +16,9 @@
...
+import UIKit
 
-public enum AccountImageSource: Sendable {
-
-    /// An image based on the provided image data is displayed.
-    case data(Data)
-
-    /// An account image based on the provided initials is generated.
-    case text(initials: String)
+public enum AccountImageSource: Equatable, Sendable {
```
Comment (samwyndham) [inline wire-ios-data-model/Source/UseCases/AccountImageSource/AccountImageSource.swift:21]: Optional: Even through it is obvious, I kind of feel it is good to update & keep the doc comments

File: wire-ios-data-model/Tests/UseCases/AccountImageSource/GetTeamAccountImageSourceUseCaseTests.swift [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            // When
+            let teamImageSource = try await sut.invoke(
+                user: user,
+                userContext: user.managedObjectContext,
+                account: coreDataStack.account
+            )
+            XCTFail("Unexpected success")
+        } catch GetTeamAccountImageSourceUseCase.Error.invalidImageSource {
+            // Then
+        }
```
Comment (samwyndham) [inline wire-ios-data-model/Tests/UseCases/AccountImageSource/GetTeamAccountImageSourceUseCaseTests.swift:150]:
```suggestion
        } catch {
            XCTFail("Unexpected error: \(error)")
        } 
```

Optional: Consider failing on default error case ()

File: wire-ios-data-model/Tests/UseCases/AccountImageSource/GetUserAccountImageSourceUseCaseTests.swift [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            // When
+            _ = try await sut.invoke(
+                user: user,
+                userContext: user.managedObjectContext,
+                account: coreDataStack.account
+            )
+            XCTFail("Unexpected success")
+        } catch GetUserAccountImageSourceUseCase.Error.invalidImageSource {
+            // Then
+        }
```
Comment (samwyndham) [inline wire-ios-data-model/Tests/UseCases/AccountImageSource/GetUserAccountImageSourceUseCaseTests.swift:114]: Same comment as in wire-ios-data-model/Tests/UseCases/AccountImageSource/GetTeamAccountImageSourceUseCaseTests.swift

**Comment by samwyndham** [comment]
@caldrian Looks good. Please check the pixelated text comment before I approve

File: WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/textMode_available.dark.png
Comment (caldrian) [reply WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/textMode_available.dark.png:1]: For better visibility in the snapshot reference image I used `CGAffineTransformScale`.
Interestingly for me is that the shapes scale so well, I would have expected everything to scale pixelated.
So I don't consider it a bug.

File: wire-ios-data-model/Source/UseCases/AccountImageSource/AccountImageSource.swift (hunk 2 above)
Comment (caldrian) [reply wire-ios-data-model/Source/UseCases/AccountImageSource/AccountImageSource.swift:21]: done

File: wire-ios-data-model/Tests/UseCases/AccountImageSource/GetTeamAccountImageSourceUseCaseTests.swift (hunk 3 above)
Comment (caldrian) [reply wire-ios-data-model/Tests/UseCases/AccountImageSource/GetTeamAccountImageSourceUseCaseTests.swift:150]: The test will fail in that case, do you still prefer to write it explicitly?

**Review by samwyndham — APPROVED** [review]

File: WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/imageMode_available.dark.png
Comment (agisilaos) [inline WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/imageMode_available.dark.png:1]: Why do we get this weird background light grey here @caldrian?

File: WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/imageMode_available.light.png
Comment (agisilaos) [inline WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/imageMode_available.light.png:1]: @caldrian Same here

**Review by agisilaos — COMMENTED** [review]
Left some questions regarding snapshots

File: WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/imageMode_available.dark.png
Comment (caldrian) [reply WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/imageMode_available.dark.png:1]: I commented this in the code. It's intentional in order to ensure, that the account image view keeps an 1:1 aspect ratio.

Comment (caldrian) [reply WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/imageMode_available.dark.png:1]: I wanted to make it visible that the green dot is aligned with the outer border of the image view, but the border of the green dot exceeds that. For me it is a bit weird and I would consider it a faulty implementation, however, it's exactly as the design specifies it.
https://www.figma.com/design/sOAbcd6UVtlZ5ejK4BoEvX/iOS?node-id=18-5795&t=uKaYX5GmTYmM6ta3-1

Comment (caldrian) [reply WireUI/Tests/WireAccountImageUITests/Resources/ReferenceImages/AccountImageViewSnapshotTests/imageMode_available.dark.png:1]: Screenshot 2024-10-28 at 9 32 27 AM

**Review by agisilaos — APPROVED** [review]
Thank you @caldrian! Looking good!

**Pull request merged**